megatron/training.py

```diff
@@ -956,11 +962,11 @@ def build_train_valid_test_data_loaders(
             [int(do_train), int(do_valid), int(do_test)])
     else:
         flags = torch.cuda.LongTensor([0, 0, 0])
-
     # Broadcast num tokens.
-    torch.distributed.broadcast(flags,
-                                mpu.get_tensor_model_parallel_src_rank(),
-                                group=mpu.get_tensor_model_parallel_group())
+    # torch.distributed.broadcast(flags, 
+    #                             mpu.get_tensor_model_parallel_src_rank(), 
+    #                             group=mpu.get_tensor_model_parallel_group())
+    torch.distributed.broadcast(flags, 0)
```

Comment:
You may start another PR for cherry-pick this. Meanwhile, you should cherry-pick the whole commit.